Tutorial beginners (#77)
nuwave/lighthouse-docs: docs/getting-started-tutorial-beginners.md

```php
'namespaces' => [
        'models' => 'App', 
        ...
],
```

Make sure to create a empty `blog_db` database and set up connection variables in  `.env` file 

```bash
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=blog_db
DB_USERNAME=username
DB_PASSWORD=password
```

Run migrations, seed database with some fake users and start the server.

```php
# create tables (DDL)
php artisan migrate

# open tinker shell
php artisan tinker

# run this command inside tinker and exit
factory('App\User', 10)->create();

# start the server
php artisan serve

```

To make sure everything is working fine, access Laravel GraphQL Playground on http://127.0.0.1:8000/graphql-playground and try some Queries:

```graphql
query {
  user(id: 1) {
    id
    name
    email
  }
}
```

Now, let's move on and create a GraphQL API for our Blog.

<br />

## The Models

One user can publish many posts, and each post has many comments from anonymous users.

<div align="center">
  <img src="/docs/assets/tutorial/model.png">  
  <p><small>Diagram</small></p>
</div>

> This is pure Laravel. After creating models and migrations remember to run migrate command:

Reviewer: ```suggestion
This first part is pure Laravel, we will add GraphQL part afterwards. Remember to run migration command after  creating models and migrations:
```

        return $this->belongsTo(Post::class);
    }
}
```

```php
<?php

use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

class CreatePostsTable extends Migration
{
    public function up()
    {
        Schema::create('posts', function (Blueprint $table) {
            $table->increments('id');
            $table->unsignedInteger('user_id');
            $table->string('title');
            $table->string('content');
            $table->timestamps();
        });
    }

    public function down()
    {
        Schema::dropIfExists('posts');
    }
}
```

```php
<?php

use Illuminate\Support\Facades\Schema;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Migrations\Migration;

class CreateCommentsTable extends Migration
{
    public function up()
    {
        Schema::create('comments', function (Blueprint $table) {
            $table->increments('id');
            $table->unsignedInteger('post_id');
            $table->string('reply');
            $table->timestamps();
        });
    }

    public function down()
    {
        Schema::dropIfExists('comments');
    }
}
```

## The Magic

> Let's edit `routes/schema.graphql` and define our blog schema, based on Eloquent Models we have created. This is mostly like a pure GraphQL schema definition, but with some Lighthouse sorcery. 

Reviewer: ```suggestion
Let's edit `routes/schema.graphql` and define our blog schema, based on the Eloquent Models we created. This is pure GraphQL schema definition (SDL), but with some Lighthouse sorcery through directives. 
```

```graphql
type User {
    id: ID!
    name: String!
    email: String!
    created_at: DateTime!
    updated_at: DateTime!
    posts: [Post] @hasMany
}

type Post {
    id: ID!
    title: String!
    content: String!
    user: User! @belongsTo
    comments: [Comment] @hasMany
}

type Comment{
    id: ID!
    reply: String!
    post: Post! @belongsTo
}

type Query{
    posts: [Post] @all
    post (id: Int! @eq): Post @find
}

```

In this example:

- Note that `type` name is the same as a Eloquent Model

> - Directives add some behavior in your schema

Reviewer: Maybe make all of these link to the official documentation also.

  - `@hasMany` / `@belongsTo` express eloquent relationships 
  - `@all` fetch all eloquent models and return the collection
  - `@find` find a model based on the arguments provided
  - `@eq` places an equal operator on a eloquent query.  

> 

Reviewer: Maybe add something like "There are tons of other directives in Ligthouse, you can read more about them here".
